src/deepsparse/loggers/function_logger.py

```diff
@@ -0,0 +1,137 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Implementation of the Function Logger
+"""
+
+from typing import Any, Dict, List, Union
+
+from deepsparse.loggers import BaseLogger, MetricCategories
+from deepsparse.loggers.metric_functions import apply_function
+
+
+__all__ = ["FunctionLogger"]
+
+
+class FunctionLogger(BaseLogger):
+    """
+    DeepSparse logger that applies functions to raw log values
+    according to the specified config file
+
+    :param loggers: A single (or list of) DeepSparse Logger object(s)
+    :param config: A configuration dictionary that specifies the mapping between
+        the target name and the functions to be applied to raw log values
+        e.g.
+        ```
+        {"pipeline_inputs":
+            [{"function": "identity",
+              "target_logger": "python", # if not specified log to all available loggers
+              "frequency": 3},
+
+            [{"function": "identity",
+              "frequency": 5}],
+
+         "pipeline_outputs":
+            [{"function": "identity",
+            "frequency": 4}]
+        }
+    """
+
+    def __init__(
+        self,
+        loggers: Union[BaseLogger, List[BaseLogger]],
+        config: Dict[str, List[Dict[str, Any]]],
+    ):
+
+        self.loggers = loggers if isinstance(loggers, list) else [loggers]
+        self.config = self._add_frequency_counter(config)
+
+    @property
+    def identifier(self) -> str:
+        return "function"
+
+    def log(self, identifier: str, value: Any, category: MetricCategories):
+        """
+        :param identifier: The identifier of the log
+             By default should consist of at least one string (name of the pipeline):
+                e.g. identifier = "pipeline_name"
+            If identifier is to be more complex, optional strings are to be concatenated
+            using dot as a separator:
+                e.g. identifier = "pipeline_name.some_argument_1.some_argument_2"
+
+            Note: by convention, "some_argument_1" very often is
+                  referred to as `target_name`
+        :param value: The data structure that the logger is logging
+        :param category: The metric category that the log belongs to
+        """
+        if category == MetricCategories.DATA:
+            pipeline_name, target_name, *_ = identifier.split(".")
+            list_functions = self.config.get(target_name)
+            if list_functions:
```

Comment:
```suggestion
            list_functions = self.config.get(target_name)
            if list_functions is None:
                # NOTE: explicitly filtering values not present in the config
                return
           # TODO & unindent everything below
```
Will make filtering more explicit and reduce identation

Reply:
Oh you like guard clauses dont you :)

Reply:
Hah you know it 🤣